### ISSUE TITLE
Keymap does not work

### ISSUE THREAD

**Issue by zongUMR** [zed-industries/community#977]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I defined customized keymap but it doesn't work.

### Environment

```
Zed: v0.73.3 (stable)
OS: macOS 13.1.0
Memory: 16 GiB
Architecture: aarch64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="894" alt="Screenshot 2023-02-20 at 11 06 12" src="https://user-images.githubusercontent.com/11454175/220000302-d0fad314-cbf4-40e6-bdb9-b5ebff68109d.png">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

Althougth the below log has an `invalid keystore ctrl-c-d` error, but I already changed it to `ctrl-c ctrl-d`, but doesn't work either.
[log.txt](https://github.com/zed-industries/community/files/10779122/log.txt)

**Comment by Kethku** [comment]
`ctrl-c` is bound to `menu::Cancel` in the default keymap, so your bindings dont have a chance to reach the second key.

This issue is a ux and understandability problem with our keybinding system. Some ways forward:
1. We must give users a way to understand what action is being dispatched by a given keypress maybe through a toggleable setitng or log
2. We should surface to the user when there are pending keys in the matcher. This would make it more obvious when a key has already been consumed
3. We must give the user a way to override these default bindings. Today later bindings match first, but this doesn't properly handle when bindings have multiple steps as shown above.

**Comment by zongUMR** [comment]
Yes exactly, I think the 3rd point is the most important for an ending user.

**Comment by zongUMR** [comment]
@Kethku Will Zed allow users to override the default bindings in the next release? So I can use `ctrl-c ctrl-d` to trigger **goToDefiniton** action, it is an important and useful shortcut for me like in other editors

**Comment by Kethku** [comment]
Unfortunately no. I will make sure to discuss this with the team to come up with a fix soon

**Comment by aexvir** [comment]
I guess this is the same thing I reported some time ago in https://github.com/zed-industries/community/issues/399
I'll close that one as it didn't manage to get any comment

**Comment by zongUMR** [comment]
> Unfortunately no. I will make sure to discuss this with the team to come up with a fix soon

Is it possible to bind `g-d` to **goToDefinition**? I tried it but did not work either.

**Comment by TENX-S** [comment]
> > Unfortunately no. I will make sure to discuss this with the team to come up with a fix soon
> 
> 
> 
> Is it possible to bind `g-d` to **goToDefinition**? I tried it but did not work either.

I'm looking for this too

**Comment by Kethku** [comment]
> > Unfortunately no. I will make sure to discuss this with the team to come up with a fix soon
> 
> Is it possible to bind `g-d` to **goToDefinition**? I tried it but did not work either.

What do you mean by this? Do you mean pressing the `g` and `d` keys together and queueing a goToDefinition action?

**Comment by zongUMR** [comment]
Press `g` and immediately with a `d` to trigger the **goToDefinition** action, it's a very common action in VIM

**Comment by zongUMR** [comment]
> Oh I see. This can be done today using some vim specific keymap contexts.
> 
> Most of the operators in vim mode will set the `vim_operator` context variable for active editors to the operator key. So for this case, you can use `vim_operator == g` to create bindings locked behind the `g` key in normal mode.
> 
> Binding `g` `d` to `goToDefinition` looks like this:
> 
> ```
> {
>     "context": "Editor && vim_operator == g",
>     "bindings": {
>         "d": "editor::GoToDefinition"
>     }
> }
> ```
> 
> Add that to your personal keymap and you should be good to go.

It works, thanks Kethku
